src/slice.rs

```diff
@@ -402,3 +413,107 @@ impl<'data, T: 'data + Send> Producer for ChunksMutProducer<'data, T> {
          })
     }
 }
+
+
+/// Parallel iterator over slices separated by a predicate
+pub struct Split<'data, T: 'data, P> {
+    slice: &'data [T],
+    separator: P,
+}
+
+struct SplitProducer<'data, 'sep, T: 'data, P: 'sep> {
+    slice: &'data [T],
+    separator: &'sep P,
+
+    /// Marks the endpoint beyond which we've already found no separators.
+    tail: usize,
+}
+
+impl<'data, T, P> ParallelIterator for Split<'data, T, P>
+    where P: Fn(&T) -> bool + Sync,
+          T: Sync
+{
+    type Item = &'data [T];
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        let producer = SplitProducer {
+            slice: self.slice,
+            separator: &self.separator,
+            tail: self.slice.len(),
+        };
+        bridge_unindexed(producer, consumer)
+    }
+}
+
+impl<'data, 'sep, T, P> UnindexedProducer for SplitProducer<'data, 'sep, T, P>
+    where P: Fn(&T) -> bool + Sync,
+          T: Sync
+{
+    type Item = &'data [T];
+
+    fn split(mut self) -> (Self, Option<Self>) {
+        let SplitProducer { slice, separator, tail } = self;
+
+        // Look forward for the separator, and failing that look backward.
```

Comment:
This is basically analogous to the split we do on strings right now? Seems very familiar. =)